refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java

```diff
@@ -116,79 +108,66 @@ private Problem reallyDoRefactoring(boolean saveAfterDone) {
+                    changeRefactoringElement(element, true);
                     current += progressStep;
                     fireProgressListenerStep((int) current);
                 }
             } finally {
-                for (Transaction commit : commits) {
-                    SPIAccessor.DEFAULT.check(commit, false);
-                }
-
-                UndoableWrapper wrapper = MimeLookup.getLookup("").lookup(UndoableWrapper.class);
-                for (Transaction commit : commits) {
-                    if (wrapper != null) {
-                        setWrappers(commit, wrapper);
+                commits.forEach(commit -> SPIAccessor.DEFAULT.check(commit, false));
+                UndoableWrapper undoableWrapper = MimeLookup.getLookup("").lookup(UndoableWrapper.class);
+                commits.forEach(commit -> {
+                    if (undoableWrapper != null) {
+                        undoableWrapper.setActive(true, this);
                     }
-
-                    if(commit instanceof ProgressProvider) {
+                    if (commit instanceof ProgressProvider) {
                         ProgressProvider progressProvider = (ProgressProvider) commit;
                         progressProvider.addProgressListener(progressListener);
-                    }
-                    try {
-                        commit.commit();
-                    } finally {
-                        if(commit instanceof ProgressProvider) {
-                            ProgressProvider progressProvider = (ProgressProvider) commit;
+                        try {
+                            commit.commit();
+                        } finally {
                             progressProvider.removeProgressListener(progressListener);
                         }
+                    } else {
+                        commit.commit();
                     }
-                    if (wrapper != null) {
-                        unsetWrappers(commit, wrapper);
+                    if (undoableWrapper != null) {
+                        undoableWrapper.setActive(false, null);
                     }
+                });
+                if (undoableWrapper != null) {
+                    undoableWrapper.close();
                 }
-                if (wrapper != null) {
-                    wrapper.close();
-                }
-                for (Transaction commit : commits) {
-                    SPIAccessor.DEFAULT.sum(commit);
-                }
+                commits.forEach(commit -> SPIAccessor.DEFAULT.sum(commit));
             }
             if (saveAfterDone) {
                 LifecycleManager.getDefault().saveAll();
-                for (DataObject dob:DataObject.getRegistry().getModified()) {
-                    SaveCookie cookie = dob.getCookie(SaveCookie.class);
+                for (DataObject dataObject : DataObject.getRegistry().getModified()) {
+                    SaveCookie cookie = dataObject.getLookup().lookup(SaveCookie.class);
```

Comment:
There was a change here:

Before: `dataObject.getCookie(SaveCookie.class)`

After: `dataObject.getLookup().lookup(SaveCookie.class)`

Reply:
getLookup().lookup(SaveCookie.class) is the proper way.

Reply:
Oh, okay. Corroborating, I see that NetBeans provides an automatic fix which rewrites the getCookie() call to `getLookup().lookup(SaveCookie.class)`.